### PR TITLE
helm: Improve linting and add log level override

### PR DESCRIPTION
**Pull request by webvictim**
Updates #5582 

- Adds a log level override to the `teleport-cluster` and `teleport-kube-agent` charts to allow users to easily change Teleport's log level to`DEBUG`. The default level is set to `INFO`.
  - Teleport is not easy to configure from scratch without `DEBUG` logs, and there is currently no way to set this other than by manually manipulating the Helm-configured Teleport `ConfigMap`, which will likely not survive chart upgrades. This isn't something easily covered by the planned "escape hatch" for mounting configs either, as you'd have to write a whole config yourself just to alter the log level.
  - @klizhentas I know you resisted this before but I think this is a necessary override while Teleport relies so much on `DEBUG` level logging.
- Improves the current `lint-helm` Makefile target to run `yamllint` on the output of `helm template` to make sure there are no glaring errors which would prevent `kubectl` parsing the output.
  - Also installs `yamllint` in the buildbox
  - Fixes the existing linter complaints.
    - We would ideally also run`teleport configure --test` with the generated Teleport config from Helm as described in https://github.com/gravitational/teleport/issues/5559 to make sure that Teleport sees the config as valid, but this feature has not been implemented yet.

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -36,14 +39,18 @@ RUN apt-get update -y --fix-missing && \
         make \
         mingw-w64 \
         net-tools \
+        openssh-client \
```
Comment (awly) [inline build.assets/Dockerfile:42]: why do we need `openssh-client`?

**Review by awly — APPROVED** [review]

File: build.assets/Dockerfile (hunk 1 above)
Comment (webvictim) [reply build.assets/Dockerfile:42]: I added `--no-install-recommends` to the `apt-get` commands in an effort to make the layer size a bit smaller; this had the side effect of no longer installing `openssh-client`, which was seemingly a recommendation included when installing another package. It was causing unit tests to fail as they need the `scp` binary.

File: Makefile [hunk 2]
```diff
@@ -358,12 +363,17 @@ lint-sh:
...
+		if [ "$${DRONE}" = "true" ]; then echo "This is a failure when running in CI." && exit 1; fi; \
+		exit 0; \
+	fi; \
 	for CHART in $$(find examples/chart -mindepth 1 -maxdepth 1 -type d); do \
 		if [ -d $$CHART/.lint ]; then \
 			for VALUES in $$CHART/.lint/*.yaml; do \
 				echo "$$CHART: $$VALUES"; \
 				helm lint --strict $$CHART -f $$VALUES || exit 1; \
-				helm template test $$CHART -f $$VALUES 1>/dev/null || exit 1; \
+				helm template test $$CHART -f $$VALUES | yamllint -d relaxed - || exit 1; \
```
[the remark this answers is not in the document]
Comment (webvictim) [reply Makefile:376]: The `default` configuration is too finicky IMO - it errors on stuff like a missing document start `---`, indentation not being exactly 4 spaces (Helm uses 2) etc. `relaxed` just insists on spacing being consistent throughout the file, ensures there's no trailing space, etc. 

I looked at extending the `default` configuration and changing the spacing config to account for Helm plus disabling other irrelevant bits, but I ended up with ~80% of the `relaxed` config so figured it was easier just to use that.

[the remark this answers is not in the document]
Comment (awly) [reply Makefile:376]: That's totally fine, consider dropping a comment about that here

[the remark this answers is not in the document]
Comment (webvictim) [reply Makefile:378]: Done!

**Review by r0mant — APPROVED** [review]

**Pull request merged**